contrib/datastore/file/gcp/gcp.go

```diff
@@ -85,27 +94,232 @@ func (g GcpAdapter) CreateFileShare(ctx context.Context, fs *file.CreateFileShar
 }
 
 func (g GcpAdapter) GetFileShare(ctx context.Context, fs *file.GetFileShareRequest) (*file.GetFileShareResponse, error) {
-	panic("implement me")
+	isCreateOpDone, err := g.isOperationDone(createOp)
+	if g.checkForOperation(ctx, isCreateOpDone, createOp, "create", err) != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	isUpdateOpDone, err := g.isOperationDone(updateOp)
+	if g.checkForOperation(ctx, isUpdateOpDone, updateOp, "update", err) != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	isDeleteOpDone, err := g.isOperationDone(deleteOp)
+	if g.checkForOperation(ctx, isDeleteOpDone, deleteOp, "delete", err) != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	instance, err := g.GetInstance(ctx, fs.Fileshare)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+	log.Debugf("Found Instance %+v", instance)
+
+	fileShare, err := g.ParseFileShare(instance)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	if instance.State == "READY" {
+		// GCP FileStore Instance exists & ready to use
+		fileShare.Status = "available"
+		return &file.GetFileShareResponse{
+			Fileshare: fileShare,
+		}, nil
+	} else {
+		// GCP FileStore Instance exists but is not ready to use
+		return &file.GetFileShareResponse{
```

Comment:
If the fileshare is not ready, it means provider is already tried to initialize or trying to make it ready, So the status should be not ready. Right? If we send error, what does the API shows as o/p

Reply:
Could you please elaborate which provider?
Else is taking care of sending the error for any STATE NOT READY for FS instance.

Reply:
GCP Cloud provider.

Reply:
Yes, GCP FS operations are submitted to the cloud provider. Now, polling takes care of checking the status of the FS instance and waits for getting it ready from cloud provider. If there is any failure in the operation, Else clause will take care of returning the Error to the service layer.
If we send error, API update the FS state to "error" & return